R/knitr-engine.R

```diff
@@ -292,6 +296,16 @@ eng_python_initialize <- function(options, envir) {
   ensure_python_initialized()
   eng_python_initialize_hooks(options, envir)
 
+  if (options$cache > 0) {
+    module <- tryCatch(import("dill"), error = identity)
+    if (inherits(module, "error")) {
+      if (module$message == "ImportError: No module named dill") {
```

Comment:
It's bad practice to test the message directly, since it might be translated to other languages in different locales. Instead, you can probably just warn if the `dill` package couldn't be loaded.

Reply:
This was completely changed. Used `py_last_error()`.